crates/biome_js_analyze/src/lint/style/no_restricted_imports.rs

```diff
@@ -386,115 +860,440 @@ impl ImportRestrictionStatus {
     PartialEq,
     Serialize,
 )]
-#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(rename_all = "camelCase", deny_unknown_fields, default)]
-pub struct CustomRestrictedImportOptions {
-    /// The message to display when this module is imported.
-    #[serde(skip_serializing_if = "str::is_empty")]
-    message: Box<str>,
+pub struct PatternOptions {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    group: Option<Box<[Box<str>]>>,
```

Comment:
We introduced our own glob type some months ago.
And the import organizer provides a wrapper named `ImportSourceGlob` dedicated to source glob matching.
Thus, I could write this:

```suggestion
    group: Option<Box<[ImportSourceGlob]>>,
```

Actually, I wonder if we should take one more step and use the same source matcher as our import organizer:

```suggestion
    group: Option<SourcesMatcher>,
```

This could allow users to use a glob or a list of globs.
Also, this could allow users to use predefined matchers like `:NODE:` (cf the import organizer for more details).

Going even farther, we could use `GroupMatcher` that supports matcher like `{ type: true, source: "@app/**" }`. I could extend it and add support for matching imported names (e.g. `{ type: true, names: ["A", "B"] source: "@app/**" }`). This could remove the need of options like `importNames`.
This requires some discussion with other team members.